tests/expectations/metrics/test_metrics_util.py

```diff
@@ -632,6 +656,115 @@ def test__eq__(
+        execution_engine=engine,
+        table_selectable=sqlalchemy.quoted_name("table_with_single_pk", quote=False),
+        schema_name=None,
+    )
+
+    assert columns is not None
+    assert len(columns) == 3  # id, name, value
+
+    # All columns should have primary_key field
+    assert all("primary_key" in col for col in columns)
+
+    # Only 'id' should be marked as primary key
+    pk_columns = [col["name"] for col in columns if col["primary_key"]]
+    assert pk_columns == ["id"]
+
+    # Other columns should not be primary keys
+    non_pk_columns = [col["name"] for col in columns if not col["primary_key"]]
+    assert set(non_pk_columns) == {"name", "value"}
+
+
+@pytest.mark.unit
+def test_get_sqlalchemy_column_metadata_composite_primary_key(sql_connector_test_engine):
+    """Test that composite primary keys are correctly identified."""
+    from great_expectations.expectations.metrics.util import get_sqlalchemy_column_metadata
+
+    engine = sql_connector_test_engine("table_with_composite_pk", "__test_composite_pk")
+
+    columns = get_sqlalchemy_column_metadata(
+        execution_engine=engine,
+        table_selectable=sqlalchemy.quoted_name("table_with_composite_pk", quote=False),
+        schema_name=None,
+    )
+
+    assert columns is not None
+    assert len(columns) == 4  # user_id, order_id, product, quantity
+
+    # All columns should have primary_key field
+    assert all("primary_key" in col for col in columns)
+
+    # Both user_id and order_id should be marked as primary keys
+    pk_columns = sorted([col["name"] for col in columns if col["primary_key"]])
+    assert pk_columns == ["order_id", "user_id"]
+
+    # Other columns should not be primary keys
+    non_pk_columns = sorted([col["name"] for col in columns if not col["primary_key"]])
+    assert non_pk_columns == ["product", "quantity"]
+
+
+@pytest.mark.unit
+def test_get_sqlalchemy_column_metadata_no_primary_key(sql_connector_test_engine):
+    """Test that tables without primary keys don't break."""
+    from great_expectations.expectations.metrics.util import get_sqlalchemy_column_metadata
+
+    engine = sql_connector_test_engine("table_without_pk", "__test_no_pk")
+
+    columns = get_sqlalchemy_column_metadata(
+        execution_engine=engine,
+        table_selectable=sqlalchemy.quoted_name("table_without_pk", quote=False),
+        schema_name=None,
+    )
+
+    assert columns is not None
+    assert len(columns) == 2  # description, amount
+
+    # All columns should have primary_key field
+    assert all("primary_key" in col for col in columns)
+
+    # No columns should be marked as primary keys
+    pk_columns = [col["name"] for col in columns if col["primary_key"]]
+    assert pk_columns == []
+
+    # All columns should have primary_key=False
+    assert all(not col["primary_key"] for col in columns)
+
+
+@pytest.mark.unit
+def test_get_sqlalchemy_column_metadata_quoted_pk_column(sql_connector_test_engine):
```

Comment:
Non-blocking, but the test for this might be more robust if you used the newer integration testing patterns in `tests/integration/data_sources_and_expectations`. This would benefit from:
* following more standard practices
* allow parameterizing the test to run against a few data sources
* improve readability (at least when viewing files in github) since including the binary .db file requires a more robust editor.
That said, I'm not sure how well it supports setting primary keys, and we don't want to introduce scope creep here. So it's a thought, but not a blocker.